src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -791,21 +779,21 @@ public static DateTime FromFileTime(long fileTime)
 
         public static DateTime FromFileTimeUtc(long fileTime)
         {
-            if (fileTime < 0 || fileTime > MaxTicks - FileTimeOffset)
+            if ((ulong)fileTime > MaxTicks - FileTimeOffset)
             {
                 throw new ArgumentOutOfRangeException(nameof(fileTime), SR.ArgumentOutOfRange_FileTimeInvalid);
             }
 
 #pragma warning disable 162 // Unrechable code on Unix
             if (s_systemSupportsLeapSeconds)
             {
-                return FromFileTimeLeapSecondsAware(fileTime);
+                return FromFileTimeLeapSecondsAware((ulong)fileTime);
             }
 #pragma warning restore 162
 
             // This is the ticks in Universal time for this fileTime.
-            long universalTicks = fileTime + FileTimeOffset;
-            return new DateTime(universalTicks, DateTimeKind.Utc);
+            ulong universalTicks = (ulong)fileTime + FileTimeOffset;
+            return new DateTime(universalTicks | KindUtc);
```

Comment:
> return new DateTime(universalTicks | KindUtc); [](start = 11, length = 47)

wouldn't this can cause throwing exception in the constructor? I think KindUtc value is greater than MaxTicks

Reply:
ignore this. I noticed the constructor that takes ulong. 

---
In reply to: [565691515](https://github.com/dotnet/runtime/pull/45479#discussion_r565691515) [](ancestors = 565691515)